src/mesh.cpp

```diff
@@ -433,70 +432,76 @@ void RegularMesh::bins_crossed(const Particle* p, std::vector<int>& bins,
     // bin.  MAX_SEARCH_ITER prevents an infinite loop.
     int search_iter = 0;
     int j;
-    while (xt::any(ijk0 < 1) || xt::any(ijk0 > shape_)) {
+    bool in_mesh = true;
+    for (int i = 0; i < n; ++i)
+      if (ijk0[i] < 1 || ijk0[i] > shape_[i]) in_mesh = false;
+    while (!in_mesh) {
       if (search_iter == MAX_SEARCH_ITER) {
         warning("Failed to find a mesh intersection on a tally mesh filter.");
         return;
       }
 
       for (j = 0; j < n; ++j) {
         if (std::fabs(u[j]) < FP_PRECISION) {
-          d(j) = INFTY;
+          d[j] = INFTY;
         } else if (u[j] > 0.0) {
-          double xyz_cross = lower_left_[j] + ijk0(j) * width_[j];
-          d(j) = (xyz_cross - r0[j]) / u[j];
+          double xyz_cross = lower_left_[j] + ijk0[j] * width_[j];
+          d[j] = (xyz_cross - r0[j]) / u[j];
         } else {
-          double xyz_cross = lower_left_[j] + (ijk0(j) - 1) * width_[j];
-          d(j) = (xyz_cross - r0[j]) / u[j];
+          double xyz_cross = lower_left_[j] + (ijk0[j] - 1) * width_[j];
+          d[j] = (xyz_cross - r0[j]) / u[j];
         }
       }
 
-      j = xt::argmin(d)(0);
+      j = std::min_element(d, d+n) - d;
       if (u[j] > 0.0) {
-        ++ijk0(j);
+        ++ijk0[j];
       } else {
-        --ijk0(j);
+        --ijk0[j];
       }
 
       ++search_iter;
+      in_mesh = true;
+      for (int i = 0; i < n; ++i)
+        if (ijk0[i] < 1 || ijk0[i] > shape_[i]) in_mesh = false;
```

Comment:
Might as well have a `break` if the condition gets hit. Also, I'm not a huge fan of a for loop with no braces, so I'd suggest:
```C++
for (int i = 0; i < n; ++i) {
  if (ijk0[i] < 1 || ijk0[i] > shape_[i]) {
    in_mesh = false;
    break;
  }
}
```